src/errors.rs

```diff
@@ -0,0 +1,33 @@
+use std::io;
+use thiserror::Error;
+
+/// Errors for Dusk PKI
+#[derive(Error, Debug)]
+pub enum Error {
+    /// Invalid Compressed Point
+    #[error("Invalid Compressed Point Provided")]
+    InvalidPoint,
+    #[error("Invalid Parameters provided to the function")]
+    /// Invalid Parameters
+    InvalidParameters,
+    /// Bad Length
+    #[error("Bad Length (expected {expected:?}, got {found:?})")]
+    BadLength {
+        /// The found length
+        found: usize,
+        /// The expected length
+        expected: usize,
+    },
+    // TODO: this should be handled better, it's too generic
```

Comment:
As I said we can convert the errors with `into()`. Maybe open an isue in `anyhow` and ask for this specific situation possible solutions would be nice.

Reply:
`into()` is not necessary, it's implicit with `?`. The issue is that you need to explicitly add the conversion `From` trait when you want to convert from (for example) anyhow / thiserror to `io::Error`. 
For the other way around, you can use `#[from]` for `thiserror`.

Notice that this problem arise only when the function signature requires `io::Error`, otherwise you could just return an `anyhow::Error`.

Reply:
Ye, I see. It's fine then. Not something we should loose time with now. Resolved.